### PR TITLE
Add nil checks for route table association state

### PR DESCRIPTION
**Pull request by blafrisch**
In aws-iso region the AssociationState isn't set in the Associations for route tables. This caused a nil pointer dereference and an RPC error when refreshing state on an `aws_route_table_association` resource.

For reference the output from the AWS CLI command `aws ec2 describe-route-tables` looks like this in AWS us-west-2:
```
{
    "RouteTables": [
        {
            "Associations": [
                {
                    "AssociationState": {
                        "State": "associated"
                    },
                    "Main": false,
                    "RouteTableAssociationId": "rtbassoc-000000000000000000",
                    "RouteTableId": "rtb-000000000000000000",
                    "SubnetId": "subnet-000000000000000000"
                }
...
```
And looks like this in an AWS ISO region:
```
{
    "RouteTables": [
        {
            "Associations": [
                {
                    "Main": false,
                    "RouteTableAssociationId": "rtbassoc-000000000000000000",
                    "RouteTableId": "rtb-000000000000000000",
                    "SubnetId": "subnet-000000000000000000"
                }
...
```

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Output from acceptance testing (chosen because of this PR [#21710](https://github.com/hashicorp/terraform-provider-aws/pull/21710)):
```
% make testacc PKG_NAME=internal/service/ec2 TESTARGS='-run=TestAccEC2RouteTable_basic\|TestAccEC2RouteTableAssociation_Subnet_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 20  -run=TestAccEC2RouteTable_basic\|TestAccEC2RouteTableAssociation_Subnet_basic -timeout 180m
=== RUN   TestAccEC2RouteTableAssociation_Subnet_basic
=== PAUSE TestAccEC2RouteTableAssociation_Subnet_basic
=== RUN   TestAccEC2RouteTable_basic
=== PAUSE TestAccEC2RouteTable_basic
=== CONT  TestAccEC2RouteTableAssociation_Subnet_basic
=== CONT  TestAccEC2RouteTable_basic
--- PASS: TestAccEC2RouteTable_basic (65.35s)
--- PASS: TestAccEC2RouteTableAssociation_Subnet_basic (78.33s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	82.046s
```

This is my first PR for this project, let me know if there's anything else I should add/update/change. Thank you!

cc: @YakDriver @ewbankkit because of other recent ISO region related changes and context.

### CODE REVIEW

File: internal/service/ec2/status.go [hunk 1]
```diff
@@ -312,6 +312,10 @@ func StatusRouteTableAssociationState(conn *ec2.EC2, id string) resource.StateRe
 			return nil, "", err
 		}
 
+		if output.AssociationState == nil {
+			return nil, "", nil
+		}
+
```
Comment (ewbankkit) [inline internal/service/ec2/status.go:318]: @blafrisch Thanks for the contribution :tada: :clap:.
This code indicates that an `AssociationState` of `nil` is equivalent to a non-existent or disassociated association - Is that the correct behavior? If there is no `AssociationState` but there is an association, is that indicative of an associated association or not?

Comment (blafrisch) [reply internal/service/ec2/status.go:318]: If there is no `AssociationState` but there is an association it is indicative of an association. Not sure of the best way to represent this as the state is not set.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc PKG_NAME=internal/service/ec2 TESTARGS='-run=TestAccEC2RouteTableAssociation_\|TestAccEC2MainRouteTableAssociation_' ACCTEST_PARALLELISM=4
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 4  -run=TestAccEC2RouteTableAssociation_\|TestAccEC2MainRouteTableAssociation_ -timeout 180m
=== RUN   TestAccEC2MainRouteTableAssociation_basic
=== PAUSE TestAccEC2MainRouteTableAssociation_basic
=== RUN   TestAccEC2RouteTableAssociation_Subnet_basic
=== PAUSE TestAccEC2RouteTableAssociation_Subnet_basic
=== RUN   TestAccEC2RouteTableAssociation_Subnet_changeRouteTable
=== PAUSE TestAccEC2RouteTableAssociation_Subnet_changeRouteTable
=== RUN   TestAccEC2RouteTableAssociation_Gateway_basic
=== PAUSE TestAccEC2RouteTableAssociation_Gateway_basic
=== RUN   TestAccEC2RouteTableAssociation_Gateway_changeRouteTable
=== PAUSE TestAccEC2RouteTableAssociation_Gateway_changeRouteTable
=== RUN   TestAccEC2RouteTableAssociation_disappears
=== PAUSE TestAccEC2RouteTableAssociation_disappears
=== CONT  TestAccEC2MainRouteTableAssociation_basic
=== CONT  TestAccEC2RouteTableAssociation_Subnet_changeRouteTable
=== CONT  TestAccEC2RouteTableAssociation_Gateway_basic
=== CONT  TestAccEC2RouteTableAssociation_disappears
--- PASS: TestAccEC2RouteTableAssociation_disappears (35.52s)
=== CONT  TestAccEC2RouteTableAssociation_Gateway_changeRouteTable
--- PASS: TestAccEC2RouteTableAssociation_Gateway_basic (43.95s)
=== CONT  TestAccEC2RouteTableAssociation_Subnet_basic
--- PASS: TestAccEC2RouteTableAssociation_Subnet_changeRouteTable (63.39s)
--- PASS: TestAccEC2MainRouteTableAssociation_basic (63.73s)
--- PASS: TestAccEC2RouteTableAssociation_Subnet_basic (36.79s)
--- PASS: TestAccEC2RouteTableAssociation_Gateway_changeRouteTable (66.05s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	108.221s
```

File: internal/service/ec2/status.go (hunk 1 above)
Comment (ewbankkit) [reply internal/service/ec2/status.go]: We can do this by returning a "fake" `RouteTableAssociationState` from `StatusRouteTableAssociationState` with a `State` value of "associated".

**Comment by ewbankkit** [comment]
@blafrisch Thanks for the contribution :tada: :clap:.

**Pull request merged**